Firestore/core/src/firebase/firestore/util/status.h

```diff
@@ -132,6 +137,10 @@ class ABSL_MUST_USE_RESULT Status {
   // a `State` structure containing the error code and message(s)
   std::unique_ptr<State> state_;
 
+  // Whether this instance has been moved. Without this `ok()` will be true
+  // for moved instances.
+  bool is_moved_from_ = false;
```

Comment:
Accessing a moved-from instance is not something we especially need to care about because you're not supposed to do it. It's perfectly reasonable (and normal) for the moved-from state to be equivalent to the default-initialized state. Indeed some in the community advocate that this is the way to do it: https://foonathan.net/blog/2016/08/24/move-default-ctor.html.

We should avoid this flag. StatusOr's destructor is making incorrect assumptions about the behavior of Status after move. The assertion is wrong and should be fixed/removed.

Reply:
As discussed offline, reusing the `State*` member to tag if it is moved.